strapi/strapi
# [V4]: Add a working example of a plugin written in Typescript

### What does it do?

This PR adds a new example (`/examples`) under the name "`typescript-plugin`".
This plugin doesn't do much for now (except exposing a service `foo` with a method `bar` that logs a message in the console.
The main goal of this plugin (and thus this PR) is to showcase an example of how to integrate a Typescript environment for a Strapi plugin.

### How to test it?

To test the plugin, transpile the JS code to the build folder using the build or develop scripts (_the develop script rebuild the sources on file change_).
Then simply link the build directory of the plugin to a Strapi V4 application using the `/config/plugins.js` file and try calling the service with the following code:

```javascript
strapi
  .plugin('typescriptplugin')
  .service('foo')
  .bar()
```

It should log a message in the console.

### Tasks

- [x] Make a working typescript plugin
- [ ] Add a readme with explanation & how to build/develop etc...
